core/src/main/scala/kafka/controller/ControllerEventManager.scala

```diff
@@ -139,4 +140,19 @@ class ControllerEventManager(controllerId: Int,
     }
   }
 
+  private def pollFromEventQueue(): QueuedEvent = {
+    val count = eventQueueTimeHist.count()
```

Comment:
Hmm, if eventQueueTimeHist.count() is not 0, it seems that we can block on queue.take() forever without resetting the histogram?

Reply:
@junrao I'm not sure I follow. If there are no more events in the queue (`count != 0`), we'd like the histogram to reset after a specified timeout. `count` will always be a non-zero value until we reset the histogram.

If we block on queue.take() forever even when `count != 0`, wouldn't that behave the same way as before? In which case do you suggest resetting?

Reply:
@jeffkbkim : Yes, you are right. The code seems fine.